### PR TITLE
Unittest to pytest channels, config, visa

### PR DESCRIPTION
**Pull request by jenshnielsen**
* test_channels
* test_config
* test_monitor (just removing an unused import)
* test_visa

* And add missing close logic that was missing in #2232

### CODE REVIEW

File: qcodes/tests/test_config.py [hunk 1]
```diff
@@ -167,102 +164,166 @@ def path_to_config_file_on_disk(tmp_path):
...
+        'current_schema',
+        new_callable=PropertyMock
+    )
+    env = mocker.patch.object(
+        Config,
+        'env_file_name',
+        new_callable=PropertyMock
+    )
+    load_config = mocker.patch.object(Config, 'load_config')
+    isfile = mocker.patch('os.path.isfile')
```
Comment (astafan8) [inline qcodes/tests/test_config.py:196]: refactor these ~4 things into shared fixture(s)?

File: qcodes/tests/test_visa.py [hunk 2]
```diff
@@ -70,104 +70,113 @@ def query(self, cmd):
...
-
-        # this one doesn't
-        inst4 = MockBackendVisaInstrument('name4',
-                                          address='ASRL4', visalib='@py')
-        self.addCleanup(inst4.close)
-        self.assertEqual(rm_mock.call_count, 4)
-        self.assertEqual(rm_mock.call_args, (('@py',),))
-        self.assertEqual(address_opened[0], 'ASRL4')
-        inst4.close()
+args1 = [
```
Comment (astafan8) [inline qcodes/tests/test_visa.py:73]:
```suggestion
 # error args for set(-10)
args1 = [
```

**Review by astafan8 — APPROVED** [review]

File: qcodes/tests/test_config.py (hunk 1 above)
Comment (jenshnielsen) [reply qcodes/tests/test_config.py:196]: ok then

Comment (jenshnielsen) [reply qcodes/tests/test_config.py:196]: There is still room for improvements but that will be another time

**Pull request merged**